test/tests/problems/eigen_problem/ne_deficient_b.i

```diff
@@ -0,0 +1,101 @@
+[Mesh]
+  type = GeneratedMesh
+  dim = 2
+  xmin = 0
+  xmax = 10
+  ymin = 0
+  ymax = 10
+  elem_type = QUAD4
+  nx = 8
+  ny = 8
+[]
+
+[Variables]
+  [./u]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+  [./v]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+[]
+
+[Kernels]
+  [./diff_u]
+    type = Diffusion
+    variable = u
+  [../]
+  [./diff_v]
+    type = Diffusion
+    variable = v
+  [../]
+
+  [./rhs]
+    type = CoupledForce
+    variable = u
+    v = v
+    eigen_kernel = true
+    coef = -1.0
+  [../]
+  [./src_v]
+    type = CoupledForce
+    variable = v
+    v = u
+  [../]
+[]
+
+[BCs]
+  [./homogeneous_u]
+    type = DirichletBC
+    variable = u
+    boundary = '0 1 2 3'
+    value = 0
+  [../]
+  [./homogeneous_v]
+    type = DirichletBC
+    variable = v
+    boundary = '0 1 2 3'
+    value = 0
+  [../]
+[]
+
+[Preconditioning]
+  [./smp]
+    type = SMP
+    full = true
+  [../]
+[]
+
+[Problem]
+  type = EigenProblem
+  eigen_problem_type = gen_non_hermitian
+  n_eigen_pairs = 1
+  n_basis_vectors = 18
+  which_eigen_pairs = TARGET_MAGNITUDE
+[]
+
+[Executioner]
+  type = Steady
+  eigen_solve_type = MONOLITH_NEWTON
+  petsc_options_iname = '-eps_power_snes_mf_operator'
+  petsc_options_value = '1'
```

Comment:
Not have to. If we have a good idea to form Jacobian, we should not use Jacobian-free.